frontend/src/metabase/query_builder/components/dataref/DatabaseSchemasPane.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint "react/prop-types": "warn" */
+import React, { useMemo } from "react";
+import { ngettext, msgid } from "ttag";
+
+import Search from "metabase/entities/search";
+import type { Card } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+import Database from "metabase-lib/lib/metadata/Database";
+import {
+  NodeListItemLink,
+  NodeListItemName,
+  NodeListItemIcon,
+  NodeListTitle,
+  NodeListContainer,
+  NodeListIcon,
+  NodeListTitleText,
+  ModelId,
+} from "./NodeList.styled";
+
+interface DatabaseSchemasPaneProps {
+  show: (type: string, item: unknown) => void;
+  database: Database;
+  models: Card[];
+}
+
+const DatabaseSchemasPane = ({
+  database,
+  show,
+  models,
+}: DatabaseSchemasPaneProps) => {
+  const sortedModels = useMemo(
+    () => models?.sort((a, b) => a.name.localeCompare(b.name)),
+    [models],
+  );
+  const schemas = database.schemas;
+  return sortedModels ? (
+    <NodeListContainer>
+      {sortedModels?.length ? (
```

Comment:
Can we do something to reduce ternary usage here a bit? It's easy to make rendering code quite difficult to follow when lots of ternaries are used. An immediate idea: seems like we don't want to render anything if `sortedModels` are nullish, so we can do something like:

```jsx
if (!sortedModels) {
  return null;
}

return ...
```

Reply:
Same comment for the `SchemaPane` component

Reply:
Same comment for the `SchemaPane` component